master/docs/manual/cfg-buildsteps.rst

```diff
@@ -2327,6 +2327,15 @@ LogRenderable
 This build step takes content which can be renderable and logs it in a pretty-printed format.
 It can be useful for debugging properties during a build.
 
+.. bb:step:: AssertRenderable
+
+AssertRenderable
+++++++++++++++++
+
+.. py:class:: buildbot.steps.master.AssertRenderable
+
+This build step takes the expression passed in as first argument. If the expression is not callable (already evaluated by the renderer) it will test if this result evaluates to ``True``. If the expression is callable it will pass in ``self.getProperties()`` and test if the result evaluates to ``True``.
+
```

Comment:
I agree that the callable functionality can be handy, but I would stick with only supporting renderable mechanism.

Another thing that would be good, and is coherent with the all-is-renderable principle is to support comparaisons for Property using __eq__ and __ne__


so that
```python
   Assert(util.Property("foo") == "bar")
```
works as expected, and generates a renderable like
```python
   Assert(util.renderer(lambda x: x.getProperty("foo") == "bar")))
```